src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionListBuilder_Sync.cs

```diff
@@ -0,0 +1,248 @@
+                        (adjustedNewPosition.Value >= textChange.Span.Start));
+
+                    changeSpan = textChange.Span;
+                    (insertText, insertTextFormat) = getPossiblySnippitizedInsertText(textChange, adjustedNewPosition);
+
+                    // If we're expecting there to be unimported types, put in an explicit sort text to put things already in scope first.
+                    // Otherwise, omit the sort text if it's the same as the label to save on space.
+                    sortText = GetSortText(completion, labelText, expectingImportedItems);
+
+                    // If the completion is replacing a bigger range than the previously-typed word, we need to have the filter
+                    // text compensate. Clients will use the range of the text edit to determine the thing that is being filtered
+                    // against. For example, override completion:
+                    //
+                    //    override $$
+                    //    |--------| Range that is being changed by the completion
+                    //
+                    // That means vscode will consider "override <additional user input>" when looking to see whether the item
+                    // still matches. To compensate, we add the start of the replacing range, up to the start of the current word,
+                    // to ensure the item isn't silently filtered out.
+
+                    if (changeSpan != typedSpan)
+                    {
+                        if (typedSpan.Start < changeSpan.Start)
+                        {
+                            // This means that some part of the currently-typed text is an exact match for the start of the
+                            // change, so chop off changeSpan.Start - typedSpan.Start from the filter text to get it to match
+                            // up with the range
+                            int prefixMatchElement = changeSpan.Start - typedSpan.Start;
+                            Debug.Assert(completion.FilterText!.StartsWith(sourceText.GetSubText(new TextSpan(typedSpan.Start, prefixMatchElement)).ToString()));
+                            filterText = completion.FilterText.Substring(prefixMatchElement);
+                        }
+                        else
+                        {
+                            var prefix = sourceText.GetSubText(TextSpan.FromBounds(changeSpan.Start, typedSpan.Start)).ToString();
+                            filterText = prefix + completion.FilterText;
+                        }
+                    }
+                    else
+                    {
+                        filterText = labelText == completion.FilterText ? null : completion.FilterText;
+                    }
+                }
+            }
+
+            static (string?, InsertTextFormat) getPossiblySnippitizedInsertText(TextChange change, int? adjustedNewPosition)
```

Comment:
I am not sure this is a real word, but if it were, it would likely be `snippetized` 🙂